glusterblockrestd/blockvolmanager/volume_manager.go

```diff
@@ -0,0 +1,144 @@
+package blockvolmanager
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"github.com/gluster/gluster-block-restapi/pkg/api"
+	"github.com/gluster/gluster-block-restapi/pkg/size"
+	"github.com/gluster/gluster-block-restapi/pkg/utils"
+)
+
+// BlockVolumeManager defines set of methods for various gluster block operation
+type BlockVolumeManager interface {
+	CreateBlockVolume(req *api.BlockVolumeCreateReq) ([]byte, error)
+}
+
+// BlockVolumeCliOptFuncs receives a blockVolumeCLI object and overrides its members
+type BlockVolumeCliOptFuncs func(cli *blockVolumeCLI)
+
+type blockVolumeCLI struct {
+	cliPath string
+}
+
+// WithCLIPath overrides cliPath of gluster-block
+func WithCLIPath(path string) BlockVolumeCliOptFuncs {
+	return func(cli *blockVolumeCLI) {
+		cli.cliPath = path
+	}
+}
+
+// NewBlockVolumeCLI returns a concrete instance implementing the BlockVolumeManager
+// interface.
+func NewBlockVolumeCLI(optFuncs ...BlockVolumeCliOptFuncs) BlockVolumeManager {
+	bm := &blockVolumeCLI{}
+
+	if defaultCliPath, err := exec.LookPath("gluster-block"); err == nil {
+		bm.cliPath = defaultCliPath
+	}
+
+	for _, optFunc := range optFuncs {
+		optFunc(bm)
+	}
+	return bm
+}
+
+// CreateBlockVolume creates a gluster block volume using gluster-block cli
+// command to create a block volume:
+// create  <volname/blockname>   [ha <count>]
+//                               [auth <enable|disable>]
+//                               [prealloc <full|no>]
+//                               [storage <filename>]
+//                               [ring-buffer <size-in-MB-units>]
+//                               <host1[,host2,...]> [size]
+// create block device [defaults: ha 1, auth disable, prealloc no, size in bytes,
+// ring-buffer default size dependends on kernel]
+func (bm *blockVolumeCLI) CreateBlockVolume(req *api.BlockVolumeCreateReq) ([]byte, error) {
+	cmdArgs := []string{"create", req.HostingVolume + "/" + req.Name}
+
+	if req.HaCount > 0 {
+		cmdArgs = append(cmdArgs, "ha", strconv.Itoa(req.HaCount))
+	}
+
+	if req.FullPrealloc {
+		cmdArgs = append(cmdArgs, "prealloc", "full")
+	}
+
+	if req.AuthEnabled {
+		cmdArgs = append(cmdArgs, "auth", "enable")
+	}
+
+	if req.RingBufferSizeInMB > 0 {
+		cmdArgs = append(cmdArgs, "ring-buffer", strconv.FormatUint(req.RingBufferSizeInMB, 10))
+	}
+
+	if req.Storage != "" {
+		cmdArgs = append(cmdArgs, "storage", req.Storage)
+	}
+
+	blockSizeInBytes := uint64(0)
+
+	switch blockSize := req.Size.(type) {
+	case string:
+		sz, err := size.Parse(blockSize)
+		if err != nil {
+			return nil, err
+		}
+		blockSizeInBytes = uint64(sz)
+	case float64:
```

Comment:
is this required? I don't think will support float size

Reply:
go unmarhsal any number(int/float) in float64 by default, since i used size as interface{} type, so in type switch i used float64